Import for vg_28xx occ

juzen2003 opened the pull request:
This pull request is to track the progress of adding VG_28xx to OPUS: (need to work with this pdsfile branch: https://github.com/SETI/pds-webtools/pull/38)
- Items done:
    - Add import steps for VG_28xx (profile index files are under Shared-OPUS/pdsdata/holdings/metadata/VG_28xx)
- To do items:
    - In OPUS, update the ring geometry values in the import for VG_28xx 
    - In pdsfile, add rules for VG_28xx
    - In OPUS, add api results test for VG_28xx

juzen2003 committed:
Include steps to import VG_2801
- In config_data.py, add VG_28xx related info to VOLSETS_WITH_PREVIEWS,
  INSTRUMENT_ABBREV_TO_MISSION_ABBREV, INSTRUMENT_ABBREV_TO_INSTRUMENT_NAME,
  VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME, VOLUME_ID_PREFIX_TO_MISSION_ABBREV,
  VOLUME_ID_ROOT_TO_TYPE, and TARGET_NAME_MAPPING. Also Create
  VOLUME_ID_TO_INSTRUMENT_NAME to store Mapping from VOLUME ID to
  instrument name for VG_28xx.
- In do_import.py (line 171-172), modify create_tables_for_import to
  fetch the correct instrument from VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME
  for VG_28xx.
- In do_import.py (line 712-716), modify import_one_index to fetch
  the correct instrument from VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME
  for VG_28xx.
- In import_util.py, update yield_import_volume_ids by adding 'VG_28xx'.
- In populate_obs_instrument_VGISS.py (line 322-327), update
  populate_obs_mission_voyager_VGISS_mission_phase_name_OBS to fetch
  MISSION_PHASE_NAME from index file.
- In populate_obs_mission_voyager.py (line 85-86), update
  populate_obs_mission_voyager_ert to handle the case when ert is not
  available.
- In populate_obs_mission_voyager.py (line 102-145), update
  populate_obs_mission_voyager_spacecraft_clock_count1/2 to handle the
  case when SPACECRAFT_CLOCK_PARTITION_NUMBER is not available.
- In populate_obs_mission_voyager.py (line 50-57), update
  helper_voyager_planet_id to fetch planet id from TARGET_NAME for VG_28xx.
- In populate_util.py (line 146-150), update _star_name_helper to get the
  star name from SIGNAL_SOURCE_NAME_1 for VG_28xx.
- In populate_util.py (line 136-142), add populate_target_name_from_index
  to fetch target name from TARGET_NAME in index file.
- In obs_general.json (line 81-84), add VG_28xx instruments to mult options
  for instrument_id.
- In obs_mission_voyager.json (line 132-133), change the data source of
  mission_phase_name from index file to function because VG_28xx doesn't
  have mission_phase_name in index file.
- Create import/populate_obs_instrument_VGPPS_occ.py to import for VG_2801.

juzen2003 committed:
Add steps to import VG_2802 (VGUVS)
- Create populate_obs_instrument_VGUVS_occ.py based on
  populate_obs_instrument_VGPPS_occ.py.

juzen2003 committed:
Add steps to import VG_2803 (RSS)
- Update populate_obs_occultation_VGRSS/VGPPS/VGUVS_source_OCC to fetch
  source from SIGNAL_SOURCE_NAME_1 in index file.

juzen2003 committed:
Update the *_wl_band_OCC & *_source_OCC to fetch wavelength band from
both WAVELENGTH_BAND_1 & WAVELENGTH_BAND_2 and occultation source from
SIGNAL_SOURCE_NAME_1 & SIGNAL_SOURCE_NAME_2.

juzen2003 committed:
Add steps to import VG_2810
- In config_data.py (line 200,209), add VG_2810 related info to
  VOLUME_ID_TO_INSTRUMENT_NAME and VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME.
- Create populate_obs_instrument_VGISS_occ.py to import VG_2810.
- In config_data.py (line 47), import populate_obs_instrument_VGISS_occ.
- In populate_obs_instrument_VGISS.py (line 334,343,353), add type to
  functions to distinguish between OBS and OCC.
- In obs_instrument_vgiss.json (line 102,130,131), add N/A as one of mult
  options for shutter_mode.
- In obs_occultation.json (line 275-282), add "Visual" as one of mult
  options for wl_band.

juzen2003 committed:
Fix the observation_type, quantity, and occ_type for VG_2810.
- In obs_occultation.json (line 79-83), add "Reflectance" as one of the
  occ_type stored in db.
- In populate_obs_instrument_VGISS_occ.py (line 202-203), return "REF"
  in populate_obs_occultation_VGISS_occ_type_OCC.
- In populate_obs_instrument_VGISS_occ.py (line 70), return "REFECT" in
  populate_obs_general_VGISS_quantity_OCC.
- In populate_obs_instrument_VGISS_occ.py (line 73-74), return "INT" in
  populate_obs_general_VGISS_observation_type_OCC.

juzen2003 committed:
Fixed import error by updating the logic in do_import.py to make sure
both EBROCC and VG_28xx instrument_name are properly handled.
- In do_import.py (line 174-179, 719), update the logic to properly
  handle instrument_name of both EBROCC and VG_28xx.
- In import_for_tests.sh, put back all volumes (including VG_28xx) for
  test import.
- In populate_obs_mission_voyager.py (line 107,109,135,137), properly
  fetch "VOLUME_ID" from index_label to determine if we should get
  spacecraft_clock_count1/2 from supplemental index or profile index files.

juzen2003 committed:
Rename all populate_obs_*_occ.py to populate_obs_*_prof.prof.py

juzen2003 committed:
Remove extra space at the return of populate_obs_instrument_*_channel_OCC

juzen2003 committed:
Remove unused functions in populate_obs_instrument_*_prof.py

juzen2003 committed:
Rename all functions in populate_*_prof.py, change the suffix from "OCC"
to "PROF". Also change "OCC" to "PROF" in VOLUME_ID_ROOT_TO_TYPE in
config_data.py.

juzen2003 committed:
Set the default widget to Instrument Name, Observation Type, and Intended
Target Name.
- In settings.py (line 308), update opus default widget to
'instrument,observationtype,target'.

juzen2003 committed:
Update all ring elevation calculations for VGISS_prof.

juzen2003 committed:
Update ring geometry calculations for VGRSS_PROF

juzen2003 committed:
Fix the calculations for ring geo in VGRSS_PROF

juzen2003 committed:
Update ring geometry calculations for VGUVS_PROF.

juzen2003 committed:
Add calculations of ring geometry in VGPPS_PROF.

juzen2003 committed:
Fixed and cleaned up variables naming in each populate_*_VG*_prof.py to
make sure import can be run without errors.

juzen2003 committed:
Remove the extra space at line 543 in populate_obs_instrument_VGRSS_prof.py

juzen2003 committed:
Update Django tests  results on default widgets, image intensity, tables
triggered by mission Voyager.

juzen2003 committed:
Add api results tests for VG_28xx, here are the list of opus ids:
- 'co-vims-occ-2014-175-l2pup-e',
- 'vg-pps-2-s-occ-1981-238-delsco-e',
- 'vg-pps-2-u-occ-1986-024-delta-sigsgr-e',
- 'vg-pps-2-n-occ-1989-236-sigsgr-i',
- 'vg-uvs-1-s-occ-1980-317-iother-e',
- 'vg-rss-2-u-occ-1986-024-four-s43-i',
- 'vg-uvs-2-n-occ-1989-236-sigsgr-i',
- 'vg-iss-2-s-prof'
(Will have to update tests when profile_index files are available)

rfrenchseti commented on opus/application/apps/results/test_results.py:
```diff
@@ -408,8 +408,7 @@ def test__get_triggered_tables_voyager(self):
                     'obs_ring_geometry',
-                    'obs_mission_voyager',
-                    'obs_instrument_vgiss']
+                    'obs_mission_voyager']
```
We have triggered table tests for each mission and each instrument. So let's add new tests for the new Voyager instruments (PPS, UVS, RSS) below after VGISS.

rfrenchseti commented on opus/import/config_data.py:
```diff
@@ -185,7 +192,21 @@
+}
+
+# Mapping from VOLUME ID to instrument name
+VOLUME_ID_TO_INSTRUMENT_NAME = {
```
Is this needed? I think it's covered in the dict above.

rfrenchseti commented on opus/import/import_util.py:
```diff
@@ -102,6 +104,8 @@ def yield_import_volume_ids(arguments):
                 elif desc.upper() == 'CORSS':
                     volume_descs.append('CORSS_8xxx')
+                elif desc.upper() == 'VOYAGER':
+                    volume_descs.append('VG_28xx')
```
I don't think this does the right thing. VOYAGER should mean VGISS and VG_28xx but it is only giving VG_28xx because of the "elif" below.

rfrenchseti replied:
Also should add options for VGUVS, VGPPS, and VGRSS.

rfrenchseti commented on opus/import/import_for_tests.sh:
```diff
@@ -24,6 +24,7 @@ python main_opus_import.py --do-all-import EBROCC_0001 $1
 python main_opus_import.py --do-all-import CORSS_8001 $1
 python main_opus_import.py --do-all-import COUVIS_8001 $1
 python main_opus_import.py --do-all-import COVIMS_8001 $1
+python main_opus_import.py --do-all-import VG_2801 VG_2802 VG_2803 VG_2810 $1
```
Maybe this should be VG_28xx instead?

rfrenchseti commented on opus/import/populate_obs_mission_voyager.py:
```diff
@@ -13,7 +13,11 @@
+    'U RINGS': 'URANUS ENCOUNTER',
+    'N RINGS': 'NEPTUNE ENCOUNTER'
+}
 ################################################################################
```
Put a blank line or two here

rfrenchseti commented on opus/import/table_schemas/obs_general.json:
```diff
@@ -564,7 +567,8 @@
               [   5, "STS", "Spectral Time Series", "060", "Y", null],
-              [   6, "OCC",  "Occultation Profile", "070", "Y", null]
+              [   6, "OCC",  "Occultation Profile", "070", "Y", null],
+              [   7, "INT",    "Intensity Profile", "080", "Y", null]
```
I think this should be "reflectance". Technically an occultation is intensity, since we're seeing how bright a star is when it's seen through a ring. Also we call it "reflectance" in the occultation type table.

rfrenchseti commented on opus/import/table_schemas/obs_occultation.json:
```diff
@@ -272,13 +273,14 @@
+              [   3, "IR",     "IR",      "040", "Y", null],
+              [   4, "VI",     "Visible", "050", "Y", null],
+              [   5, "UV",     "UV",      "060", "Y", null],
+              [   6, "VISUAL", "Visual",  "070", "Y", null],
```
I'm not sure what "VISUAL" means. Isn't this the same as "Visible"? For light visible to the naked eye.

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,616 @@
+################################################################################
+# populate_obs_instrument_VGUVS_occ.py
```
Filename at the top of this and all other _occ files is wrong - should be _prof

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,616 @@
+    assert inst_host in ['VOYAGER 1', 'VOYAGER 2']
+
+    return 'VG'+inst_host[-1]
+    # return 'VG'
```
Remove

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,616 @@
+def populate_obs_occultation_VGUVS_optical_depth_max_PROF(**kwargs):
+    return None
+
+def _integration_duration_helper(**kwargs):
```
I don't think this is used, since there is no INTEGRATION_DURATION column in the supp index.

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,616 @@
+    if wl_band2 == 'N/A':
+        wl_band = wl_band[0]
+
+    return wl_band
```
I don't understand this function. The return value has to be either a single string, or a tuple of strings where one is what we put in the database and the other is what we display to the user next to a checkbox. But it looks like you could be returning [wl_band1, wl_band2], which is a pair of strings that have nothing to do with each other, so what you put in the database and what you display to the user are unrelated. We can't have two wavelength bands for a single observation; we have to choose one. Are there actually cases where the bands are different?

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,616 @@
+    else:
+        src_name = [src_name1, src_name2]
+
+    return src_name
```
Same issue as above - we can't have two different sources for a single observation. If they're different, we have to pick one. I can't imagine they are ever different, though, are they?

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,616 @@
+# Source IOTA HER is at north, observer Voyager is at south. Target: S ring.
+
+# Incidence angle: the angle between the point where incoming source photons
+# hit the ring, to the north pole of the earth (normal vector
```
Not north pole of the earth... the pole on whatever planet you're looking at that's on the lit side of the rings.

juzen2003 committed:
Update the return of populate_obs_occultation_*_source_PROF and
populate_obs_occultation_*_wl_band_PROF. Now they will return a string.

rfrenchseti commented on opus/import/populate_obs_instrument_VGISS.py:
```diff
@@ -319,14 +319,19 @@ def populate_obs_occultation_VGISS_host_OBS(**kwargs):
 # THESE NEED TO BE IMPLEMENTED FOR EVERY VOYAGER INSTRUMENT
 ################################################################################
 
-# There is nothing instrument-specific for Voyager.
```
This file should be renamed to _obs.

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,611 @@
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    wl2 = index_row['MAXIMUM_WAVELENGTH']
+
```
Remove blank line to be consistent with wavelength1

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,611 @@
+    return 'NONE'
+
+
+### populate_obs_occultation TABLE ###
```
This comment was previously wrong in all the files - can you fix it? It should say "### OBS_OCCULTATION TABLE ###"

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,611 @@
+def populate_obs_ring_geometry_VGUVS_ring_intercept_time2_PROF(**kwargs):
+    return populate_time1_from_index(column='RING_EVENT_STOP_TIME', **kwargs)
+
+def _is_Voyager_at_north(**kwargs):
```
Move this above first use so it's easier to find. Also don't capitalize Voyager - we don't normally have capital letters in function names unless it's something special like an abbreviation.

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,611 @@
+
+# Opening angle to observer: the angle between the ring surface to the direction
+# where outgoing photons to the observer. Positive if observer is at the north
+# side of the ring , negative if it's at the south side. If observer is at the
```
Extra space before comma

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,611 @@
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+    if _is_Voyager_at_north(**kwargs):
+        target_name = populate_target_name_from_index(**kwargs)
```
How about a new function called something like _is_voyager_at_north_except_uranus. Then you don't have to hard-code the target name stuff in every function here and you can remove one level of IFs as well.

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,611 @@
+    index_row = metadata['index_row']
+    src_name = index_row['SIGNAL_SOURCE_NAME_1']
+    target_name = index_row['TARGET_NAME'].upper().strip()
+    if (src_name == "DELTA SCO"
```
You can just return the result of the comparison without any IF

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,611 @@
+
+
+################################################################################
+# THESE NEED TO BE IMPLEMENTED FOR EVERY voyager INSTRUMENT
```
Another bad search-and-replace that might be historical but you can fix. Make VOYAGER all caps in this comment in all relevant files.

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,565 @@
+    assert inst_host in ['VOYAGER 1', 'VOYAGER 2']
+
+    return 'VG'+inst_host[-1]
+    # return 'VG'
```
Remove

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,565 @@
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    wl2 = index_row['MAXIMUM_WAVELENGTH']
+
```
Remove

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,565 @@
+def populate_obs_occultation_VGRSS_optical_depth_max_PROF(**kwargs):
+    return None
+
+def _integration_duration_helper(**kwargs):
```
Remove

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,565 @@
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+
+    target_name = populate_target_name_from_index(**kwargs)
```
Use new function per UVS

rfrenchseti commented on opus/import/populate_obs_instrument_VGPPS_prof.py:
```diff
@@ -0,0 +1,604 @@
+    assert inst_host in ['VOYAGER 1', 'VOYAGER 2']
+
+    return 'VG'+inst_host[-1]
+    # return 'VG'
```
Remove

rfrenchseti commented on opus/import/populate_obs_instrument_VGPPS_prof.py:
```diff
@@ -0,0 +1,604 @@
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    wl2 = index_row['MAXIMUM_WAVELENGTH']
+
```
Remove

rfrenchseti commented on opus/import/populate_obs_instrument_VGPPS_prof.py:
```diff
@@ -0,0 +1,604 @@
+    index_row = metadata['index_row']
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+    if _is_Voyager_at_north(**kwargs):
```
New function

rfrenchseti commented on opus/import/populate_obs_instrument_VGISS_prof.py:
```diff
@@ -0,0 +1,537 @@
+    assert inst_host in ['VOYAGER 1', 'VOYAGER 2']
+
+    return 'VG'+inst_host[-1]
+    # return 'VG'
```
Remove

rfrenchseti reviewed (changes requested):
- I reviewed UVS first so look there for more complete comments that are applicable to the other instruments.
- I think the obs_occultation table should be renamed to obs_profile to be consistent. When doing the search-and-replace be careful not to replace things improperly. It's best to use case-matching so you don't change comments. Then do a second replace on OBS_OCCULTATION to OBS_PROFILE with case-matching.
- You need to rename "Occultation Constraints" to whatever it was we had decided on, perhaps "Occultation&Reflectance Profile Constraints". Unfortunately I can't find my notes but perhaps you wrote it down.

juzen2003 committed:
Fixed yield_import_volume_ids by adding options for VGPPS/VGUVS/VGRSS.
- In test__results.py (line 426-460), add triggered table tests on VGPPS
  , VGUVS, and VGRSS. (will have to fix this later)
- In config_data.py (line 203), remove unused dictionary.
- In import_util.py (line 108-124), add options options for VGPPS/VGUVS/
  VGRSS in yield_import_volume_ids.

juzen2003 committed:
Fixed the mults options for wl_band in obs_occultation.json and observation_type
in obs_general.json.
- In obs_general.json (line 571), change the "intensity Profile" to
  "Reflectance".
- In obs_occultation.json(line 282), remove "VISUAL", use "VISIBLE".
- In populate_obs_instrument_VGISS_prof.py (line 255-256), return "VI"
  (visible) for the wavelength band.

juzen2003 committed:
- Removed commented out codes in populate_obs_general_*_inst_host_id_PROF.
- Removed unused function _integration_duration_helper.
- Updated comments in populate_obs_ring_geometry_*_incidence1_PROF.
- Renamed populate_obs_instrument_VGISS.py to populate_obs_instrument_VGISS_obs.py

juzen2003 committed:
Clean up code styles in populate_obs_instrument_*_prof.py.

juzen2003 committed:
Create a new function _is_voyager_at_north_except_uranus to pass into
functions that calculate ring elevations.
- In populate_obs_instrument_VGPPS_prof.py (line 371-378) and
  populate_obs_instrument_VGUVS_prof.py (line 375-382), create
  _is_voyager_at_north_except_uranus to calculate ring elevations.

juzen2003 committed:
Rename "obs_occultation" table to "obs_profile"

juzen2003 committed:
Rename obs_occultation.json to obs_profile.json.

juzen2003 committed:
Renamed "Occultation Constraints" to "Occultation&Reflectance Profile Constraints".

juzen2003 committed:
- Renamed "Occultation Constraints" to "Ring/Limb Profile Constraints".
- Change the observation type of VG_2810 to "Reflectance Profile".

juzen2003 committed:
Update test results for VG_28xx.
- In create_test_results.py (line 70-76), update opus_id for VG_28xx due
  to the change of rules in pdsfile.
- Create new test results based on new opus ids for VG_28xx.

rfrenchseti commented on opus/application/test_api/test_results_contents.py:
```diff
@@ -676,6 +676,78 @@ def test__results_contents_co_vims_occ_2014_175_l2pup_e_files(self):
+        "[test_results_contents.py] vg-iss-2-s-prof files"
+        url = "/api/files/vg-iss-2-s-prof.json"
+        self._run_json_equal_file(url, "results_vg_iss_2_s_prof_files.json")
+    ####################
```
Add black line above ######

rfrenchseti commented on opus/application/test_api/test_results_contents.py:
```diff
@@ -995,3 +1067,40 @@ def test__results_contents_co_vims_occ_2014_175_l2pup_e_images(self):
+        "[test_results_contents.py] vg-iss-2-s-prof images"
+        url = "/api/images.json?opusid=vg-iss-2-s-prof"
+        self._run_json_equal_file(url, "results_vg_iss_2_s_prof_images.json")
+    ####################
```
Add blank line above #######

rfrenchseti commented on opus/import/import_util.py:
```diff
@@ -102,11 +104,24 @@ def yield_import_volume_ids(arguments):
+                    volume_descs.append('VGISS_7xxx')
+                    volume_descs.append('VGISS_8xxx')
+                    volume_descs.append('VG_28xx')
                 elif desc.upper() == 'VOYAGER' or desc.upper() == 'VGISS':
```
The comparisons with 'VOYAGER' here and below don't do anything because it was already caught above.

rfrenchseti commented on opus/import/populate_obs_mission_voyager.py:
```diff
@@ -43,7 +48,15 @@ def helper_voyager_planet_id(**kwargs):
-    mp = index_row['MISSION_PHASE_NAME']
+
+    index_label = metadata['index_label']
+    # For occulation, we will fetch data from profile index
```
This comment doesn't say anything useful. How about "For a profile, we derive the mission phase from the target name"?

rfrenchseti commented on opus/import/table_schemas/obs_general.json:
```diff
@@ -79,19 +79,22 @@
-              [  19,  "PAL200CIRC", "Palomar 200-in Cass IR", "200", "Y", null]
+              [   7,       "VGPPS",            "Voyager PPS", "080", "Y", null],
+              [   8,       "VGUVS",            "Voyager UVS", "090", "Y", null],
+              [   9,       "VGRSS",            "Voyager RSS", "100", "Y", null],
```
VGRSS should be after VGPPS to be in alphabetical order

rfrenchseti commented on opus/import/config_data.py:
```diff
@@ -69,6 +69,7 @@
                          'HSTUx_xxxx',
                          'NHxxLO_xxxx',
                          'NHxxMV_xxxx',
+                         'VG_28xx',
```
I don't think this should be here since VG_28xx doesn't have preview images

rfrenchseti commented on opus/import/config_data.py:
```diff
@@ -132,7 +133,10 @@
+    'VGISS':     'VG',
+    'VGPPS':     'VG',
+    'VGUVS':     'VG',
+    'VGRSS':     'VG',
```
Move VGRSS after VGPPS for alphabetical order

rfrenchseti commented on opus/import/config_data.py:
```diff
@@ -150,6 +154,9 @@
     'VGISS':       'Voyager ISS',
+    'VGPPS':       'Voyager PPS',
+    'VGUVS':       'Voyager UVS',
+    'VGRSS':       'Voyager RSS',
```
Move VGRSS after VGPPS for alphabetical order

rfrenchseti commented on opus/application/apps/help/templates/help/gettingstarted.html:
```diff
@@ -42,7 +42,7 @@ <h3>Searching</h3>
-            <li><b>Occultation Constraints</b> permits easy searching on
+            <li><b>Ring/Limb Profile Constraints</b> permits easy searching on
                 the common attributes of both spacecraft-based and ground-based
                 occultation profiles.</li>
```
Say "occultation and reflectance profiles"

rfrenchseti commented on opus/application/apps/paraminfo/models.py:
```diff
@@ -98,7 +98,8 @@ def body_qualified_label_results(self, referred=False):
         if (pretty_name in ['General', 'PDS', 'Wavelength', 'Image',
-                            'Occultation', 'Surface'] and not referred):
+                            'Ring/Limb Profile', 'Surface']
+                            and not referred):
```
"and not referred" should be indented under "pretty_name"

rfrenchseti reviewed (changes requested):
Many opus product types do not have tooltip descriptions. At least these:

vgpps_occ_best
vgpps_occ_02
vgpps_occ_05
vgpps_occ_raw
vgpps_occ_cal
vgpps_occ_geo
vgpps_occ_img
vgpps_occ_jit
vgpps_occ_j2000
vgpps_occ_ring
vgpps_occ_test
vgrss_occ_inv0_05
vgrss_occ_inv0_5
vgrss_occ_sam0_025
vgrss_occ_sam0_05
vgrss_occ_sam0_1
vgrss_occ_sam0_2
vgrss_occ_sam0_5
vguvs_occ_full_res
vguvs_occ_sampled_2
vguvs_occ_sampled_3
vguvs_occ_sampled_4
vguvs_occ_sampled_5
vguvs_occ_02
vguvs_occ_05
vguvs_occ_10
vguvs_occ_coadd
vguvs_occ_spec
vguvs_occ_cal
vguvs_occ_geo
vguvs_occ_img

juzen2003 committed:
Update based on github feedback.
- Change "Ring/Limb Profile" to "Occultation and Reflectance Profiles".
- Add tooltips for new opus_type in internal_def_product_types.json.

juzen2003 committed:
Chang "Occultation and reflectance Profiles" to "Occultation/Reflectance Profiles".

juzen2003 committed:
Add tooltip for vgrss_occ_inv05 & vgrss_occ_inv0_4 and re-organize the
order of product type in internal_def_product_types.json.

juzen2003 committed:
Update the wording of product types tooltips in internal_def_product_types.json.
